### ISSUE TITLE
worker.onmessage is not executed

### ISSUE THREAD

**Issue by ftarlao** [audreyt/node-webworker-threads#61]
Hi,
first, thankyou for your module, it has the best interface ever.. but I have a problem, in my code I have this (inside a foreach):
.........
if(l !== null){
        var worker = new Worker(function(){
 			this.onmessage = function(event){	
	   			var level = event.data.level;
			    	console.log(level);  								    				
				var p = dowork(level);  //dowork defined first
				console.log(p);
				postMessage(s);	
				console.log("Processing level: "+level);
    				self.close();		
	}});
       worker.onmessage = function(event) {
            console.log("I'm here");
	    pile.push(event.data);
       };
       worker.postMessage({'level':l.level});
}
......

worker.onmessage is not receiving the message from worker on the Master thread, but all the logging tell me that worker performed it's task correctly (receives data from main thread and so on).. and closes.

Is there a bug, or am I missing something?

**Comment by audreyt** [comment]
Did you attach `worker` to somewhere that's visible outside the scope? I ask because it's possible that the Worker instance gets garbage-collected before it got a chance of processing the `postMessage`.

**Comment by ftarlao** [comment]
I'm sorry for being late, I have not received a notification from github and meanwhile we have moved ahead; my colleague has solved using another tool but I remind we have tried lots of changes. Perhaps we have forgotten to bind the worker to an outside scope array..preventing gc, this makes thanks. I have not the previous code in order to check things. Thanks again for your library and for your feedback, and sorry again  for having missed your answer.
Regards
